README.md

```diff
@@ -1 +1,516 @@
+  - `'onSubmit'`: Validate when the submit is triggered
+- `revalidateMode`: String to indicate when inputs with errors get re-validated, default is `'onChange'`.
+  - `'onblur'`: Validate when an input field is blurred
+  - `'onChange'`: Validate when an input field value changes
+  - `'onSubmit'`: Validate when the submit is triggered
+- `submitFocusError`: Boolean, when `true` focus on the first field with error after submit validates, if there is any. Defaults to `true`.
+- `onSubmit`: Callback function that the form values are passed to when submit is triggered
+- `noNativeValidate`: Boolean, when `true` disables the default browser validation (notice that react-hook-form-jsonschema does NOT yet implement validation form URIs and email addresses).
+
+## Hooks API
+
+The following are the common fields returned in the object from every `use'SomeInputType'` hook:
+
+- `type`: The type of the input, as defined in **`InputTypes`**:
+  - `generic`: the default type, a non specialized type, only contains the common fields
+  - `radio`: Type used for `<input type='radio' \>`
+  - `select`: Type used for `<select>`
+  - `input`: Type used for generic `<input \>`
+  - `textArea`: Type used for `<textarea>`
+- `path`: Path in the jsonschema this input is validated against. The path is always in the form: `#/some/child/data/field/here` where `#` represents the root of the schema, and the `some/child/data/field/here` represents the tree of objects (from `some` to `here`) to get to the desired field, which in this case is `here`.
+- `name`: The last object/data field name in the tree. In the case of `#/some/child/data/field/here` the name value will be `here`.
+- `isRequired`: indicates wether the field is required or not.
+- `formContext`: If you want to access internal react-hook-form context use this
+- `getError()`: Returns an `ErrorMessage`, which has the following format:
+  - `{message: ErrorTypes, expected: ErrorMessageValues}`
+  - **`ErrorTypes`**, is an enum, with the following values:
+    - `required`: the field is required to be filled
+    - `maxLength`: maximum lenght of string input was surpassed
+    - `minLength`: minimum lenght of string input was not met
+    - `maxValue`: maximum value of number input was surpassed
+    - `minValue`: minimum value of number input was not met
+    - `pattern`: the pattern or type defined in the schema was not met
+    - `multipleOf`: the number is not a multiple of the number defined in the schema
+    - `undefinedError`: the error type could not be defined
+  - **`ErrorMessageValues`**, is the expected value to be met, it will be `true` for required, and the minimum value expected for `minValue` for example.
+- `getObject()`: Returns the data field in the schema that this input refers to
+
+**Please notice that in all of the examples bellow it is assumed that the components are already children of a FormContext component**
+
+### useHidden(path)
+
+##### Description
```

Comment:
I don't think this level of heading is needed here. The description could simply follow the `###` title